server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -2266,7 +2328,8 @@ private Set<DataSegment> insertSegments(
       Set<DataSegment> segments,
       @Nullable SegmentSchemaMapping segmentSchemaMapping,
       Map<SegmentId, SegmentMetadata> upgradeSegmentMetadata,
-      Map<SegmentId, SegmentId> newVersionForAppendToParent
+      Map<SegmentId, SegmentId> newVersionForAppendToParent,
+      Map<String, String> upgradedFromSegmentIdMap
```

Comment:
Please deduplicate these two maps. They serve the same purpose.

Reply:
One of them is for append segments exclusively.
The other is for both append and replace segments

Reply:
Ah, I see. Thanks for the clarification!